java/src/main/native/src/row_conversion.cu

```diff
@@ -187,6 +189,62 @@ struct batch_data {
   std::vector<row_batch> row_batches; // information about each batch such as byte count
 };
 
+rmm::device_uvector<size_type> build_string_row_offsets(table_view const &tbl,
+                                                        size_type fixed_width_and_validity_size,
+                                                        rmm::cuda_stream_view stream) {
+  auto const num_rows = tbl.num_rows();
+  rmm::device_uvector<size_type> d_row_offsets(num_rows, stream);
+
+  std::vector<strings_column_view::offset_iterator> string_columns_offsets;
+  auto column_ptrs = thrust::make_transform_iterator(
+      tbl.begin(), [](auto const &col) -> strings_column_view::offset_iterator {
+        if (!is_fixed_width(col.type())) {
+          CUDF_EXPECTS(col.type().id() == type_id::STRING, "only string columns are supported!");
+          strings_column_view scv = col;
+          return scv.offsets_begin();
```

Comment:
Don't need to create a new object, since the pointer of `offsets_begin` points to the original column data and will still be valid.
```suggestion
          return strings_column_view(col).offsets_begin();
```